react-forms/src/TextArea.tsx

```diff
@@ -18,63 +18,74 @@
 //  ----------------------------------------------------------------------------------------------------
 //  01/22/2020 - Billy Ernest
 //       Generated original version of source code.
+//  11/03/2023 - C Lackner
+//       Added internal state to avoid cursor jumping.
 //
 // ******************************************************************************************************
 import * as React from 'react';
 import { CreateGuid } from '@gpa-gemstone/helper-functions'
 import HelperMessage from './HelperMessage';
 
 interface IProps<T> {
-    Rows: number;
-    Record: T;
-    Field: keyof T;
-    Setter: (record: T) => void;
-    Valid: (field: keyof T) => boolean;
-    Label?: string;
-    Feedback?: string;
-    Disabled?: boolean;
-    Help?: string | JSX.Element;
+  Rows: number;
+  Record: T;
+  Field: keyof T;
+  Setter: (record: T) => void;
+  Valid: (field: keyof T) => boolean;
+  Label?: string;
+  Feedback?: string;
+  Disabled?: boolean;
+  Help?: string | JSX.Element;
 }
 
 export default function TextArea<T>(props: IProps<T>) {
-    const [guid, setGuid] = React.useState<string>("");
-    const [showHelp, setShowHelp] = React.useState<boolean>(false);
+  const [guid, setGuid] = React.useState<string>("");
+  const [showHelp, setShowHelp] = React.useState<boolean>(false);
+  const [internal, setInternal] = React.useState<boolean>(false);
+  const [heldVal, setHeldVal] = React.useState<string>('');
+  
+  React.useEffect(() => {
+    if (!internal) {
+      setHeldVal(props.Record[props.Field] == null ? '' : (props.Record[props.Field] as any).toString());
+    }
+    setInternal(false);
+   }, [props.Record[props.Field]]);
 
-    React.useEffect(() => {
-        setGuid(CreateGuid());
-    }, []);
+  React.useEffect(() => {
+      setGuid(CreateGuid());
+  }, []);
 
-    const showLabel = props.Label !== "";
-    const showHelpIcon = props.Help !== undefined;
-    const label = props.Label === undefined ? props.Field : props.Label;
+  function valueChange(value: string) {
+    setInternal(true);
```

Comment:
should internal/setInternal be a useref? As it stands, we have no guarantee that this setter will go before the props.Setter. They should get batched, but I don't think we use internal to hook into any dependencies anyway.

Reply:
good catch. We do use this extensively tho so we may need to fix that on `Input` too

Reply:
taken care of in `TextArea`. also swapped `guid` since it should never change after initial call